Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -424,6 +424,51 @@ namespace AzToolsFramework
             return newTemplateId;
         }
 
+        void PrefabSystemComponent::UpdateTemplateFilePath(TemplateId templateId, const AZ::IO::PathView& filePath)
+        {
+            auto findTemplateResult = FindTemplate(templateId);
+            if (!findTemplateResult.has_value())
+            {
+                AZ_Error(
+                    "Prefab", false,
+                    "Template associated by given Id '%llu' doesn't exist in PrefabSystemComponent.",
+                    templateId);
+                return;
+            }
+
+            if (!filePath.IsRelative())
+            {
+                AZ_Error(
+                    "Prefab", false,
+                    "Provided filePath %s must be relative.", filePath.Native().data());
+                return;
+            }
+
+            Template& templateToChange = findTemplateResult->get();
+            if (templateToChange.GetFilePath() == filePath)
+            {
+                return;
+            }
+            
+            m_templateFilePathToIdMap.erase(templateToChange.GetFilePath());
+            if (!m_templateFilePathToIdMap.try_emplace(filePath, templateId).second)
+            {
+                AZ_Error("Prefab", false, "Provided filePath %s already exists.", filePath.Native().data());
+                return;
+            }
+            m_templateFilePathToIdMap.emplace(AZStd::make_pair(filePath, templateId));
+
+            PrefabDom& prefabDom = templateToChange.GetPrefabDom();
+            PrefabDomValueReference pathReference = Prefab::PrefabDomUtils::FindPrefabDomValue(prefabDom, "Source");
+            if (pathReference)
+            {
+                const AZStd::string_view pathStr = filePath.Native();
+                pathReference->get().SetString(pathStr.data(), prefabDom.GetAllocator());
```

Comment:
As `pathStr` is not guaranteed to be null terminated it's safer to pass the string length to `SetString`:
`pathReference->get().SetString(pathStr.data(), aznumeric_caster(pathStr.length()). prefabDom.GetAllocator());`